src/network/networkd-dhcp4.c

```diff
@@ -583,6 +586,58 @@ static int dhcp4_set_hostname(Link *link) {
         return sd_dhcp_client_set_hostname(link->dhcp_client, hn);
 }
 
+static bool check_promote_secondaries(const char *ifname) {
+        char *promote_secondaries_sysctl = NULL, *promote_secondaries_path = NULL;
```

Comment:
Both initializations are not needed, because those two variables are always assigned before use. We don't assign stuff when it's unconditionally overwritten (there are exceptions, but none that apply here).